src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/StatelessDataController.cs

```diff
@@ -72,66 +94,112 @@ public async Task<ActionResult> Post([FromQuery] string dataType)
                 return BadRequest($"Invalid dataType {dataType} provided. Please provide a valid dataType as query parameter.");
             }
 
-            object appModel = _altinnApp.CreateNewAppModel(classRef);
+            if (GetPartyHeader(HttpContext).Count > 1)
+            {
+                return BadRequest($"Invalid party. Only one allowed");
+            }
+
+            int? partyId = await GetPartyId(HttpContext);
+
+            EnforcementResult enforcementResult = await AuthorizeAction(org, app, partyId.Value, "read");
```

Comment:
could partyId.Value result in a null reference if value is null? 
I guess it would make sense to check if we have a count different from 1 for partyHeader to ensure that we have a party to retrieve at all

Reply:
If no header, we will pick the value from authenticated user. As now. SO I don't think it should ever be null.  We need it to authorize. But added a forbidden if null for some strange reason